docs/src/hooks/useMdxPageTableofContents.ts

```diff
@@ -0,0 +1,25 @@
+import { useStaticQuery, graphql } from "gatsby";
+
+const useMdxPageTableofContents = (pagename: string) => {
+  const { mdxPages } = useStaticQuery(
+    graphql`
+      query MdxPagesQuery {
+        mdxPages: allMdx(filter: { fields: { collection: { eq: "pages" } } }) {
+          nodes {
+            tableOfContents(maxDepth: 4)
```

Comment:
Like you said, this can only detect titles statically, in order to detect _all_ headings we would need to analyze the resulting HTML. The easiest way might be to do it on the client using [rehype-dom](https://github.com/rehypejs/rehype-dom).

Reply:
You also brought up the idea of React Context. 🤷

Reply:
I'll deal with this in a separate PR.